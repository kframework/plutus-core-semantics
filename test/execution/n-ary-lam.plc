(program

(module Foo (import) (export () ())
  (define r1 [(lam x y y) 1 2])
  (define r2 [(lam x y z (builtin addInteger 10 z)) 1 2 3])
  (define r3 [(lam x y (builtin addInteger x y)) 10 20])
  (define r4 [(lam x y z (builtin addInteger x (builtin addInteger y z))) 10 20 5])
  (define r5 [[(lam f g [(lam x x) f]) (lam x y y) (lam x y x)] #A #B]))

)
