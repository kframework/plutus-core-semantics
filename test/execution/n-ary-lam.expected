<T>
  <k>
    .
  </k>
  <lenv>
    .Map
  </lenv>
  <genv>
    Foo . r1 |-> 2
    Foo . r2 |-> 13
    Foo . r3 |-> 30
    Foo . r4 |-> 35
    Foo . r5 |-> #B
  </genv>
  <modules>
    ModuleCellSetItem ( <Module>
      <curr>
        0
      </curr>
      <name>
        Foo
      </name>
      <dataCons>
        .TyExpList
      </dataCons>
      <decl>
        .QualCList
      </decl>
      <importedModules>
        .ModNmList
      </importedModules>
      <exportTy>
        .TyExpList
      </exportTy>
      <exportTm>
        .QualCList
      </exportTm>
    </Module> )
  </modules>
</T>
